Sources/BuilderIO/Components/BuilderComponentProtocol.swift

```diff
@@ -3,9 +3,31 @@ import SwiftyJSON
 
 
 protocol BuilderViewProtocol: View {
-    var responsiveStyles: [String: String]? { get set }
-    init(options: JSON, styles: [String: String]?);
+    var componentType: BuilderComponentType { get }
+    var block : BuilderBlock { get }
+    init(block: BuilderBlock);
     
+    
+}
+
+extension BuilderViewProtocol {
+     func getFinalStyle(responsiveStyles: BuilderBlockResponsiveStyles?) -> [String: String] {
```

Comment:
Does `?` at the end of `BuilderBlockResponsiveStyles?` mean its optional? If yes, then line 15 should also handle the `nil` case.